spec.bs

```diff
@@ -464,6 +465,16 @@ The <dfn>pathname options</dfn> is an [=options=] [=struct=] with [=options/deli
   1. Return false.
 </div>
 
+<div algorithm>
+  To determine if a <dfn>hostname pattern is an IPv6 address</dfn> given a [=/pattern string=] |input|:
+
+  1. If |input|'s [=string/code point length=] is less than 2, then return false.
+  1. If |input|[0] is U+005B (`[`), then return true.
+  1. If |input|[0] is U+007B (`{`) and |input|[1] is U+005B (`[`), then return true.
+  1. If |input|[0] is U+005C (<code>\</code>) and |input|[1] is U+005B (`[`), then return true.
```

Comment:
So, the indexing syntax is not really well-defined for strings. If it were defined, it would probably refer to code units, not code points, since the primary definition is as a sequence of code units.

However, I think it's still OK and unambiguous to use here, because everything is ASCII. That is why I haven't mentioned it before. But since you asked for some focus on Unicode stuff in particular on this review, I thought I'd mention it.

If you really want to be cautious, probably the right phrasing would be something like "Let input0 be the 0th code point in input" and "Let input1 be the 1st code point in input", then work with those. (Although even there it's not well defined whether code points are 0-indexed or 1-indexed in such prose phrases. But by saying 0th we're making an implicit claim that they're 0-indexed, so it's pretty clear to the reader.)

/cc @annevk as an FYI about things to improve in Infra.

Reply:
As discussed on chat, switch this to use:

`Let |input code points| be |input| interpreted as a [=list=] of [=/code points=].`

And then:

`|input code points|[0]`